core/org.openjdk.jmc.flightrecorder.rules/src/main/java/org/openjdk/jmc/flightrecorder/rules/util/RulesToolkit.java

```diff
@@ -1223,6 +1273,34 @@ public static String toRatioPercentString(IQuantity antecedent, IQuantity conseq
 		return resultFutures;
 	}
 
+	/**
+	 * For rules that are not evaluated (due to event availability map mismatch, or missing
+	 * dependencies), create and return a completed future IResult that can be used for the rules
+	 * report.
+	 * 
+	 * @param rule
+	 *            rule to create an evaluation error result for
+	 * @param preferences
+	 *            provider of configuration values used by the ResultBuilder
+	 * @return a completed future containing an evaluation error result
+	 */
+	private static Future<IResult> evaluationErrorResult(IRule rule, IPreferenceValueProvider preferences) {
```

Comment:
I think it might be slightly confusing that this is labeled with "error" if the rule evaluation did not fail, just did not take place. It could probably just be a N/A result instead, which we should already have methods for in RulesToolkit.